src/wp-includes/load.php

```diff
@@ -239,6 +239,49 @@ function wp_get_environment_type() {
 	return $current_env;
 }
 
+/**
+ * Retrieves the current runtime environment.
+ *
+ * The value can be set via the `WP_RUNTIME_ENVIRONMENT` global system variable,
+ * or a constant of the same name.
+ *
+ * The only value currently supported is 'virtualbox'. If not set, the value
+ * defaults to an empty string.
+ *
+ * @since 6.0.0
+ *
+ * @return string The current runtime environment.
+ */
+function wp_get_runtime_environment() {
+	static $current_runtime_env = '';
+
+	if ( ! defined( 'WP_RUN_CORE_TESTS' ) && $current_runtime_env ) {
+		return $current_runtime_env;
+	}
+
+	$wp_runtime_environments = array( 'virtualbox' );
+
+	// Check if the runtime environment variable has been set, if `getenv` is available on the system.
+	if ( function_exists( 'getenv' ) ) {
+		$has_runtime_env = getenv( 'WP_RUNTIME_ENVIRONMENT' );
+		if ( false !== $has_runtime_env ) {
+			$current_runtime_env = $has_runtime_env;
+		}
+	}
+
+	// Fetch the runtime environment from a constant, this overrides the global system variable.
+	if ( defined( 'WP_RUNTIME_ENVIRONMENT' ) ) {
+		$current_runtime_env = WP_RUNTIME_ENVIRONMENT;
+	}
```

Comment:
`WP_RUNTIME_ENVIRONMENT` constant overrides the `getenv`. A little refactoring can be a micro-optimization and readability improvement.

```suggestion
	// Fetch the runtime environment from a constant, this overrides the global system variable.
	if ( defined( 'WP_RUNTIME_ENVIRONMENT' ) ) {
		$current_runtime_env = WP_RUNTIME_ENVIRONMENT;
	} elseif ( function_exists( 'getenv' ) ) {
		// Check if the runtime environment variable has been set, if `getenv` is available on the system.
		$has_runtime_env = getenv( 'WP_RUNTIME_ENVIRONMENT' );
		if ( false !== $has_runtime_env ) {
			$current_runtime_env = $has_runtime_env;
		}
	}
```

Reply:
Now using `is_virtualbox()` and filter if code not able to detect VB.